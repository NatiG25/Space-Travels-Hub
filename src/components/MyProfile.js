import React from 'react';

const MyProfile = () => (
  <>
    <h1>MyProfile</h1>
  </>
);

export default MyProfile;
