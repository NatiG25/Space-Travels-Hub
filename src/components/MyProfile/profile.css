.profileContainer {
  display: flex;
  flex-direction: row;
  width: 90%;
  height: auto;
  margin: 40px auto;
  gap: 50px;
  text-align: left;
}

h2 {
  margin-left: 40px;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

.missionContainer,
.rocketContainer {
  width: 45%;
}

li {
  list-style-type: none;
  min-height: 30px;
  border: 2px solid #bcbcbc;
  margin: 20px auto;
  padding: 10px 0 0 10px;
}
