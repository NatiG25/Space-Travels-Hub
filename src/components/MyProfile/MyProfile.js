import React from 'react';
import { useSelector } from 'react-redux';
import ProfileMission from './ProfileMission';
import './profile.css';

const MyProfile = () => {
  const reservedMissions = useSelector((state) => state.missionReducer);
  const rocketInfo = useSelector((state) => state.rocketReducer);
  const reserveRockets = rocketInfo.filter((rocket) => rocket.reserved);

  return (
    <div className="profileContainer">
      <div className="missionContainer">
        <h2>My Missions</h2>
        <ul>
          {reservedMissions
            .filter((data) => data.reserved)
            .map((filterData) => (
              <ProfileMission key={filterData.mission_id} missionName={filterData.mission_name} />
            ))}
        </ul>
      </div>
      <div className="rocketContainer">
        <h2>My Rockets</h2>
        <ul>
          {reserveRockets.map((rocket) => (
            <li key={rocket.id}>
              {rocket.rocket_name}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
