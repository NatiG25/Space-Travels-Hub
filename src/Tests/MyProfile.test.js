import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MyProfile from '../components/MyProfile/MyProfile'
import store from '../components/redux/configureStore'

describe('Test all MyProfile components and functions', () => {  
  it('renders MyProfile correctly', () => {
      const tree = renderer
        .create(<Provider store={store}><MyProfile /></Provider>)
        .toJSON();
      expect(tree).toMatchSnapshot();
  });
})
