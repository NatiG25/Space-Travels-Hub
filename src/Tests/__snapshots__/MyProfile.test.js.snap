// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test all MyProfile components and functions renders MyProfile correctly 1`] = `
<div
  className="profileContainer"
>
  <div
    className="missionContainer"
  >
    <h2>
      My Missions
    </h2>
    <ul />
  </div>
  <div
    className="rocketContainer"
  >
    <h2>
      My Rockets
    </h2>
    <ul />
  </div>
</div>
`;
